### PR TITLE
Add "sample-plugin" staging repo

### PR DESCRIPTION
**Pull request by juanvallejo**
**Release note**:
```release-note
Adds sample-cli-plugin staging repository
```

Showcases usage of the new `kubectl` plugins mechanism.
Showcases usage of the new `cli-runtime` repo from a third-party plugin

See README.md for details on what this plugin actually does.
TODO (will address in future):
- add support for `--user` flag
- some tests would be nice

cc @soltysh @smarterclayton @sttts 

/sig cli

### CODE REVIEW

**Review by soltysh — APPROVED** [review]
/lgtm
/approve

File: hack/.golint_failures [hunk 1]
```diff
@@ -722,6 +722,7 @@ staging/src/k8s.io/sample-apiserver/pkg/apiserver
 staging/src/k8s.io/sample-apiserver/pkg/cmd/server
 staging/src/k8s.io/sample-apiserver/pkg/registry/wardle/fischer
 staging/src/k8s.io/sample-apiserver/pkg/registry/wardle/flunder
+staging/src/k8s.io/sample-cli-plugin/pkg/cmd
```
Comment (sttts) [inline hack/.golint_failures]: fix the golint warning instead.

**Comment by sttts** [comment]
Please add a block to hack/import-restrictions.yaml.

File: hack/.golint_failures (hunk 1 above)
Comment (juanvallejo) [reply hack/.golint_failures]: Thanks, addressed all golint warnings

**Pull request closed without merging**